File: case_studies/Pgrowth/scripts/samples_distributions.jl
using PyPlot, Statistics, Distributions, Random

#################################################################

x = collect(-4:0.1:4);
pdfx = pdf.(Normal(0,1),x);
fig, ax = PyPlot.subplots()
ax.scatter(x,pdfx)
ax.set_ylabel("pdfx",fontsize=16)
ax.set_xlabel("x",fontsize=16);

#################################################################

sampx = rand(Normal(0,1),500);
fig, ax = PyPlot.subplots()
ax.scatter(collect(1:1:500),sampx)
ax.set_ylabel("sampx",fontsize=16);

#################################################################

fig, ax = PyPlot.subplots()
ax.hist(sampx,bins=10)
ax.set_ylabel("freq",fontsize=16)
ax.set_xlabel("sampx",fontsize=16);

#################################################################

using StatsPlots

#################################################################

StatsPlots.plot(density(sampx),xlabel = "beta1",ylabel = "Density")

File: case_studies/Pgrowth/scripts/samples_distributions.py
import scipy.stats as stats
import matplotlib.pyplot as plt

x = np.arange(-4,4.1,0.1)
pdfx = stats.norm.pdf(x)

fig, ax = plt.subplots()
ax.plot(x, pdfx, marker='o', linestyle='none')
ax.set(xlabel='x', ylabel='pdf')

####################################################################

sampx = stats.norm.rvs(size=500, loc=0, scale=1)

# an alternative is
# sampx = numpy.random.normal(size=500, loc=0, scale=1)

fig, ax = plt.subplots()
ax.plot(sampx, marker='o', linestyle='none')

#####################################################################

fig, ax = plt.subplots()
ax.hist(sampx)

#####################################################################

ksd = stats.gaussian_kde(sampx)

fig, ax = plt.subplots()
ax.plot(x, ksd(x))
ax.set(xlabel='x', ylabel='density')

